voip_utils/sip.py

```diff
@@ -190,87 +270,203 @@ def datagram_received(self, data: bytes, addr):
+                call_id = headers["call-id"]
+                ack_lines = [
+                    f"ACK {caller_endpoint.uri} SIP/2.0",
+                    f"Via: SIP/2.0/UDP {local_endpoint.host}:{local_endpoint.port}",
+                    f"From: {local_endpoint.sip_header}",
+                    f"To: {to_header}",
+                    f"Call-ID: {call_id}",
+                    "CSeq: 50 ACK",
+                    "User-Agent: test-agent 1.0",
+                    "Content-Length: 0",
+                ]
+                ack_text = _CRLF.join(ack_lines) + _CRLF
+                ack_bytes = ack_text.encode("utf-8")
+                self.transport.sendto(ack_bytes, (caller_ip, caller_sip_port))
+
+                # The call been answered, proceed with desired action here
+                local_rtp_port = self._get_call_rtp_port(call_id)
+                self.on_call(
+                    CallInfo(
+                        caller_endpoint=caller_endpoint,
+                        caller_rtp_port=remote_rtp_port,
+                        server_ip=remote_rtp_ip,
+                        headers=headers,
+                        opus_payload_type=opus_payload_type,  # Should probably update this to eventually support more codecs
+                        local_rtp_ip=local_endpoint.host,
+                        local_rtp_port=local_rtp_port,
+                    )
+                )
+            elif method == "bye":
+                # Acknowlege the BYE message when the remote party hangs up
+                _LOGGER.debug("Received BYE message: %s", message)
+                if self.transport is None:
+                    _LOGGER.debug("Skipping message: %s", message)
+                    return
+
+                # Acknowledge the BYE message, otherwise the phone will keep sending it
+                (
+                    protocol,
+                    code,
+                    reason,
+                    headers,
+                    body,
+                ) = self._parse_sip_reply(message)
+                _LOGGER.debug(
+                    "Parsed response protocol=%s code=%s reason=%s headers=[%s] body=[%s]",
+                    protocol,
+                    code,
+                    reason,
+                    headers,
+                    body,
+                )
+                rtp_info = get_rtp_info(body)
+                remote_rtp_port = rtp_info.rtp_port
+                opus_payload_type = rtp_info.payload_type
+                via_header = headers["via"]
+                from_header = headers["from"]
+                to_header = headers["to"]
+                callid_header = headers["call-id"]
+                cseq_header = headers["cseq"]
+                # We should remove the call from the outgoing calls dict now if it is there
+                self._end_outgoing_call(callid_header)
+                ok_lines = [
+                    "SIP/2.0 200 OK",
+                    f"Via: {via_header}",
```

Comment:
You can skip a bunch of local variables if you extract the value from `headers` with a single quote:
```suggestion
                    f"Via: {headers['via']}",
```